htmresearch/algorithms/sparse_net.py

```diff
@@ -0,0 +1,352 @@
+# ----------------------------------------------------------------------
+# Numenta Platform for Intelligent Computing (NuPIC)
+# Copyright (C) 2016, Numenta, Inc.  Unless you have an agreement
+# with Numenta, Inc., for a separate license for this software code, the
+# following terms and conditions apply:
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero Public License version 3 as
+# published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+# See the GNU Affero Public License for more details.
+#
+# You should have received a copy of the GNU Affero Public License
+# along with this program.  If not, see http://www.gnu.org/licenses.
+#
+# http://numenta.org/licenses/
+# ----------------------------------------------------------------------
+
+"""
+Implementation of Bruno Olshausen's sparse coding algorithm.
+It relies on the formulation developed by Olshausen and Field (1996), slightly
+modified so that it uses a Locally Competitive Algorithm (LCA) to compute the
+coefficients, rather than a vanilla gradient descent, as proposed by Rozell
+et al. (2008).
+
```

Comment:
Is it possible to briefly summarize the algorithm here (in addition to the reference)?